Homework4

Raider992 opened the pull request:
В общем, вроде всё работает, но есть два вопроса. 
Первый, при обработке исключений линтер ругается на throw и пишет "'throw' of exception caught locally". Что он имеет в виду? 
Второй на счёт моего метода удаления в задаче с корзиной. Собственно, вопрос, допустимо ли переприсваивать ссылки на массивы, являющиеся свойствами объекта, в подобном случае, или лучше было удалить элемент непосредственно из данного массива?

Raider992 committed:
Коммит первого дз по основам js

Raider992 committed:
Коммит второго дз по основам js

Raider992 committed:
Коммит второго дз по основам js

Raider992 committed:
Коммит третьего дз по основам js

Raider992 committed:
Коммит третьего дз по основам js
Добавлено третье решение к задаче номер 1

Raider992 committed:
Нулевой коммит четвёртого дз по основам js

Raider992 committed:
Коммит четвёртого дз по основам js

Om1X commented on task1.js:
```diff
@@ -0,0 +1,57 @@
+    try {
+        if (isNaN(n)) throw new Error('Ведённый аргумент не является числом');
+    } catch (e) {
+        throw e.message;
```
у вас есть блок catch где и нужно ловить ошибки, а вы в нем наоборот инициируете ошибку.

Raider992 replied:
Понял

Om1X commented on task3.js:
```diff
@@ -0,0 +1,61 @@
+    },
+
+    deleteItemByName(name) {
+        return this.items = this.items.filter(el => el.name !== name);
```
плохо так делать, у нас есть место, где лежат товары и мы можем только добавлять или удалять элементы, а не переопределять каждый раз.

Raider992 replied:
понял, перепишу метод

Raider992 committed:
Коммит исправленного дз №4 по основам JS

Исправлено:

task1:
- убрал отлов ошибок через try/catch, заменил на if.
- отказался от отдельных функций получения данных и разбиения числа на разряды, теперь функция запуска скрипта получает данные и отлавливает ошибки

task3:
- заменил переопределение корзины фильтром на удаление одного элемента с заданным именем через splice
